extensions/render/src/org/sbml/jsbml/ext/render/XMLTools.java

```diff
@@ -224,5 +246,32 @@ public static String encodeArrayShortToString(Short[] array) {
     }
     return temp;
   }
+  
+  public static boolean canDecodeStringToArrayUnsignedInt(String value) {
+    String[] array = value.split(", ");
+    for (String s : array) {
+      try { Integer.parseUnsignedInt(s); } catch(NumberFormatException e) { return false; };
+    }
+    return true;
+  }
+  
+
+  /**
+   * (not quite object-oriented style)
+   * Helper to add an entry to the {@link JSBML.INVALID_XML}-{@link XMLNode} on given TreeNode
+   * @param t to which to add the UserObject/where to modify the userObject
+   * @param attributeName
+   * @param invalidValue
+   */
+  public static void addToInvalidXMLUserObject(TreeNodeWithChangeSupport t, String attributeName, String invalidValue) {
+    if(!t.isSetUserObjects() || !t.containsUserObjectKey(JSBML.INVALID_XML)) {
+      // TODO 2020/03: this is somewhat abusive code
```

Comment:
@DEVetter Why abusive code ? 
In fact, there was already a method to do that, in org.sbml.jsbml.xml.parsers.AbstractReaderWriter.processInvalidAttribute(String, String, String, String, Object).
You should use this method instead of your method in XMLTools. You could call .AbstractReaderWriter.processInvalidAttribute from XMLTools if you think it would be easier to find (but this should be in the user/developer guide in fact).

Reply:
Forget about that, I have change the way it will work slightly and I will make the changes myself.